### PR TITLE
feat(security): add support for security.txt

### LINKED ISSUE #14315: [Feature]: support for security.txt
### Describe the feature

Just like there is a standard for ``robot.txt`` there now also is a standard for providing information to security researches on how you wish to receive security vulnerabilities.

This can be provided in a ``security.txt`` for more information https://securitytxt.org/

The contents could be configured on the website.

### Is your feature request related to a problem?

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeabakker**
fixes #14315

### CODE REVIEW

File: actions/admin/security/securitytxt.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	'csaf' => false,
+];
+
+foreach ($fields as $name => $required) {
+	$value = get_input($name);
+	if ($required && empty($value)) {
+		return elgg_error_response(elgg_echo('error:missing_data'));
+	}
+	
+	elgg_save_config("securitytxt_{$name}", $value);
```
Comment (jdalsem) [inline actions/admin/security/securitytxt.php:24]: can you first validate all fields before saving them?

File: engine/classes/Elgg/Config.php [hunk 2]
```diff
@@ -95,6 +95,15 @@
  * @property bool          $remove_branding 						Is Elgg branding disabled
  * @property int           $remove_unvalidated_users_days			The number of days after which unvalidated users will be removed
  * @property bool          $require_admin_validation
+ * @property string        $securitytxt_acknowledgments             Security.txt link to acknowledgments
```
Comment (jdalsem) [inline engine/classes/Elgg/Config.php:98]: variable name should have each word separated by an underscore... `$security_txt_*`

File: actions/admin/security/securitytxt.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?php
```
Comment (jdalsem) [inline actions/admin/security/securitytxt.php:1]: rename file to security_

File: views/default/admin/security/securitytxt.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
```
Comment (jdalsem) [inline views/default/admin/security/securitytxt.php:1]: rename to security_txt

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: actions/admin/security/securitytxt.php (hunk 1 above)
Comment (jeabakker) [reply actions/admin/security/securitytxt.php]: done

File: engine/classes/Elgg/Config.php (hunk 2 above)
Comment (jeabakker) [reply engine/classes/Elgg/Config.php]: done

File: views/default/admin/security/securitytxt.php (hunk 4 above)
Comment (jeabakker) [reply views/default/admin/security/securitytxt.php]: done

**Review by jdalsem — APPROVED** [review]

**Pull request merged**